rust/src/lib/dns.rs

```diff
@@ -261,7 +261,7 @@ pub(crate) fn reselect_dns_ifaces(
             .unwrap_or_default(),
         find_ifaces_in_desire(true, &desired.interfaces)
             .or_else(|| {
-                find_ifaces_with_static_gateways(
+                find_ifaces_with_valid_for_dns(
```

Comment:
nit: What about `find_valid_ifaces_for_dns`? I think the `with` is confusing on the function name.